### PR TITLE
fix: wrong or missing analytics request source

### PR DESCRIPTION
**Pull request by andreahaku**
## **Description**
Fixes missing analytics data for events:
- `DAPP_TRANSACTION_COMPLETED`
- `SIGNATURE_APPROVED`
- `CONTRACT_ADDRESS_COPIED`
- `APPROVAL_PERMISSION_UPDATED`
- `APPROVAL_STARTED`
- `SEND_TRANSACTION_STARTED`
- `SEND_TRANSACTION_COMPLETED`

when there's an error in getting the analytics data, the methods used to return an empty object `{}`. This PR tries to send at least some basic data that is fundamental for analytics of events while waiting for the fix of the method and a better understanding of what's really causing the error.

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Views/confirmations/Approval/index.js [hunk 1]
```diff
@@ -341,24 +348,18 @@ class Approval extends PureComponent {
...
-          ? AppConstants.REQUEST_SOURCES.SDK_REMOTE_CONN
-          : this.originIsWalletConnect
-          ? AppConstants.REQUEST_SOURCES.WC
-          : AppConstants.REQUEST_SOURCES.IN_APP_BROWSER,
         is_smart_transaction: shouldUseSmartTransaction,
         ...smartTransactionMetricsProperties,
       };
     } catch (error) {
-      return {};
+      return baseParams;
```
Comment (NicolasMassart) [inline app/components/Views/confirmations/Approval/index.js:362]: S: If you want to know what's causing this error you may add a Logger.error here to have it sent to Sentry?

File: app/components/Views/confirmations/SendFlow/Confirm/index.js [hunk 2]
```diff
@@ -326,24 +348,15 @@ class Confirm extends PureComponent {
...
-          : AppConstants.REQUEST_SOURCES.IN_APP_BROWSER,
-
-        is_smart_transaction: shouldUseSmartTransaction,
+        ...baseParams,
         ...smartTransactionMetricsProperties,
       };
     } catch (error) {
-      return {};
+      // Log the error and return the baseParams
+      console.error('Error in getAnalyticsParams:', error);
```
Comment (NicolasMassart) [inline app/components/Views/confirmations/SendFlow/Confirm/index.js:358]: S: same a s previous one, if you want to discover what is the error and can't see it happen in dev, maybe use the Logger to have it on Sentry. Then later once known and fixed, remove these logs.

File: app/components/Views/confirmations/components/ApproveTransactionReview/index.js [hunk 3]
```diff
@@ -546,46 +546,60 @@ class ApproveTransactionReview extends PureComponent {
...
       // Send analytics params to parent component so it's available when cancelling and confirming
-      onSetAnalyticsParams && onSetAnalyticsParams(params);
+      if (onSetAnalyticsParams) {
+        onSetAnalyticsParams(params);
+      }
 
       return params;
     } catch (error) {
-      return {};
+      console.error('Error in getAnalyticsParams:', error);
```
Comment (NicolasMassart) [inline app/components/Views/confirmations/components/ApproveTransactionReview/index.js:601]: S: same as above

File: app/components/Views/confirmations/components/PersonalSign/PersonalSign.tsx [hunk 4]
```diff
@@ -83,30 +83,36 @@ const PersonalSign = ({
...
+    const chainId = selectChainId(store.getState());
+    const fallbackUrl = 'N/A';
+
+    let urlHost = fallbackUrl;
+    if (pageInfo.url) {
+      try {
+        const url = new URL(pageInfo.url);
+        urlHost = url.host || fallbackUrl;
+      } catch (error) {
+        console.error('Error processing analytics parameters:', error);
```
Comment (NicolasMassart) [inline app/components/Views/confirmations/components/PersonalSign/PersonalSign.tsx:97]: S: this one is interesting as there's not a lot of ways to fail the `new URL`... So Logger could help here too.

File: app/util/confirmation/signatureUtils.js [hunk 5]
```diff
@@ -23,31 +23,40 @@ export const getAnalyticsParams = (
...
-      dapp_host_name: url && url?.host,
-      chain_id: getDecimalChainId(chainId),
-      signature_type: signType,
-      version: messageParams?.version,
-      ...pageInfo?.analytics,
-      ...blockaidParams,
-    };
   } catch (error) {
-    return {};
+    console.error('Error processing analytics parameters:', error);
```
Comment (NicolasMassart) [inline app/util/confirmation/signatureUtils.js:56]: S: same here to use Logger to find the error

**Review by omridan159 — DISMISSED** [review]

**Review by elefantel — DISMISSED** [review]
LGTM

**Review by omridan159 — APPROVED** [review]

**Review by elefantel — APPROVED** [review]

**Review by NicolasMassart — APPROVED** [review]
Looks good to me.

**Pull request merged**